hr_holidays_leave_auto_approve/models/hr_leave.py

```diff
@@ -21,15 +28,15 @@ def create(self, values):
             HrLeave, self.with_context(
                 tracking_disable=tracking_disable)
             ).create(values)
-        if res.can_approve and res.holiday_status_id.auto_approve:
-            res.action_approve()
+        res.apply_auto_approve_policy()
         return res
 
     @api.model
     def _get_auto_approve_on_creation(self, values):
         auto_approve = False
         if values.get('holiday_status_id'):
-            auto_approve = self.env['hr.leave.type'].browse(
+            leave_type = self.env['hr.leave.type'].browse(
                 values.get('holiday_status_id')
-            ).auto_approve
+            )
+            auto_approve = leave_type.auto_approve_policy != 'no'
```

Comment:
Extract as `def _should_auto_approve` method

Reply:
Don't you think we can just delete the `_get_auto_approve_on_creation()` function? It is used to disable the creation message in case the leave can be auto approved, which I dont undersand why. IMHO it is not necessary to do that

Reply:
If we're self-approving, we don't want message about user approved own request being sent to himself, BUT there can be other watchers by default. Probably if "only user will receive message" then don't send it, or exclude self from followers, something like that. We can address that as separate PR